Allow user to set alternative platform using ALT_PLATFORM env var.

cmfcruz opened the pull request:
Gives the user more flexibility in setting an alternative platform if the device is detected as not supported by the project.

Raspberry Pi 4's detected as `arm-unknown-linux-gnueabihf` can be set to `aarch64-unknown-linux-gnu`.

The script checks for the ALT_PLATFORM environment variable and sets the variable in the cc_config.xml file.

List of platforms: https://boinc.berkeley.edu/trac/wiki/BoincPlatforms

Closes #11 .

Change-type: patch
Signed-off-by: Carlo Miguel Cruz <carloc@balena.io>

chrisys commented:
@cmfcruz were you receiving this message? We should default it to a working value rather than relying on the user to specify if so, but I haven't witnessed the issue myself (yet)

cmfcruz commented:
@chrisys , I did get this message when I first connected to Rosetta@Home.  I'll try it out again without setting the alternative platform and see if the device platform `arm-unknown-linux-gnueabihf` is now supported.

cmfcruz committed:
Sets an alternative platform if the device platform is unsupported.

Some devices encounter an error saying that their device is unsupported
by the project.  Setting an alternative platform such as
aarch64-unknown-linux-gnu allows a 64bit device to participate.

Change-type: patch
Signed-off-by: Carlo Miguel Cruz <carloc@balena.io>

cmfcruz commented:
@chrisys , seeing that the project requires 64-bit devices, we can set `aarch64-unknown-linux-gnu` as a default alternative platform in case their device platform is detected by the project as unsupported.  I just included the config file for this.  No more special variables needed.

tmigone commented:
Lgmt, tested in Pi3, Pi4 and Jetson Nano

chrisys commented:
@balena-ci retest

chrisys commented:
@balena-ci retest

chrisys commented:
@nazrhom could you help me with this PR? I think it was trying to process during the downtime; it looks good but statuses maybe haven't been updated.

mbalamat commented:
@balena-ci retest

chrisys commented:
@balena-ci rebase
